packages/tailwindcss/src/index.ts

```diff
@@ -372,8 +372,11 @@ async function parseCss(
     substituteAtApply(ast, designSystem)
   }
 
-  // Replace `theme()` function calls with the actual theme variables.
-  if (css.includes(THEME_FUNCTION_INVOCATION)) {
+  // Replace `theme()` function calls with the actual theme variables. Plugins
+  // could register new rules that include functions, and JS config files could
+  // also contain functions or plugins that use functions so we need to evaluate
+  // functions if either of those are present.
+  if (css.includes(THEME_FUNCTION_INVOCATION) || plugins.length > 0 || configs.length > 0) {
```

Comment:
The `theme()` calls from utilities are *not* processed by this call to `substituteFunctions`. The call in `build()` handles that because the generated nodes are the ones that have theme().

This is purely required for the work were we're taking the config stuff and backfilling some values in the theme (which ultimately makes it into the "static" AST).

So maybe the comment could use some clarification.

Reply:
Though theme calls inside `@utility` would be processed here but that happens to be more of an optimization.

Reply:
Yeah, this is only for detecting wether `addBase` pushed stuff into the input CSS AST, right? The rest will exist only in the `designSystem`.